hal/ia32/console-serial.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * Serial console
+ *
+ * Copyright 2017, 2021 Phoenix Systems
+ * Author: Michał Mirosław, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+
+
+/* UART registers */
+enum {
+	RBR = 0, /* Receiver Buffer Register */
+	THR = 0, /* Transmitter Holding Register */
+	DLL = 0, /* Divisor Latch LSB */
+	IER = 1, /* Interrupt Enable Register */
+	DLM = 1, /* Divisor Latch MSB */
+	IIR = 2, /* Interrupt Identification Register */
+	FCR = 2, /* FIFO Control Register */
+	LCR = 3, /* Line Control Register */
+	MCR = 4, /* Modem Control Register */
+	LSR = 5, /* Line Status Register */
+	MSR = 6, /* Modem Status Register */
+	SPR = 7, /* Scratch Pad Register */
+};
+
+
+struct {
+	void *base;
```

Comment:
This should be volatile. Remove casts to volatile

Reply:
Fixed.